src/include/miopen/logger.hpp

```diff
@@ -300,7 +318,14 @@ std::ostream& LogParam(std::ostream& os, std::string name, const std::vector<T>&
             miopen_log_func_ss << miopen::LoggingPrefix() << "}" << std::endl;          \
             std::cerr << miopen_log_func_ss.str();                                      \
         }                                                                               \
-    while(false)
+        if(miopen::IsLoggingToRoctx())                                                  \
+        {                                                                               \
+            std::ostringstream miopen_log_func_ss;                                      \
```

Comment:
Just a suggestion. Shouldn't the function name combining be unified with `if(miopen::IsLoggingFunctionCalls())`?
I mean all that `__FUNCTION__` \ `__PRETTY_FUNCTION__` and so on.
It's just an idea and you can just resolve it without any comments or changes.

Reply:
Yeah.  The function name and formatting are very different.  The arguments into LogParam are different.  The way it is there are only those two outer guard checks.  I like that better than having guard checks on each place the delimiter and args are added.  And I mostly left the original code in place line for line, less chance of me breaking it. :)